src/core/StelCore.cpp

```diff
@@ -1395,83 +1410,98 @@ double StelCore::getUTCOffset(const double JD) const
 		qWarning() << "Invalid timezone: " << tzName;
 	}
 
-	qint64 shiftInSeconds = 0;
-	if (tzName=="system_default" || (loc.planetName=="Earth" && !tzValid && !QString("LMST LTST").contains(tzName)))
+	// Complete the cache object and hash it.
+	u.loc= & const_cast<StelLocation &>(loc);
```

Comment:
This is some ugliness here. Why do you keep the object by `const` reference and then violate this constness?

Reply:
This is what various warnings made me do.

Reply:
Well, this is the wrong way to read the warnings then :P They are supposed to make you improve the code, not equip it with kludges.